ethstore/src/accounts_dir/disk.rs

```diff
@@ -104,8 +105,16 @@ pub type RootDiskDirectory = DiskDirectory<DiskKeyFileManager>;
 
 /// Disk directory key file manager
 pub trait KeyFileManager: Send + Sync {
+	/// Read `SafeAccount` from given key file stream with password
+	#[allow(unused_variables)]
+	fn read_with_password<T>(&self, filename: Option<String>, reader: T, password: &Option<Password>)
+		-> Result<SafeAccount, Error> where T: io::Read {
+		unimplemented!()
```

Comment:
this is a badly designed interface. what you should do is:

```rust
pub struct KeyFileManagerWithPassword {
  password: Password,
}

impl KeyFileManager for KeyFileManagerWithPassword {
  fn read<T>(&self, filename: Option<String>, reader: T) -> Result<SafeAccount, Error> where T: io::Read;
}
```

Reply:
thx, will rethink it